msprime/intervals.py

```diff
@@ -33,83 +33,232 @@ class RateMap:
     """
     A class mapping a numeric value to a set of adjacent intervals along the
     genome.
+
+    :param list position: A list of :math:`n+1` positions, starting at 0, and ending
+        in the sequence length over which the RateMap will apply.
+    :param list rate: A list of :math:`n` positive rates that apply between each
+        position.
+    :param float start_position: When returning a mean_rate over the genome, ignore the
```

Comment:
I wonder if there's much point in these start_position and end_position values, actually. Since we only use them in mean_rate calculations, wouldn't it be simpler to add an ``exclude_telomeres=True`` option there, which would disregard any flanking zero-rate regions?

It's not really helping the user at the moment, since they have to decide where the start and end positions of the map are, presumably doing the same numpy calculations as we'd do to find them.

Reply:
They get passed in when reading a hapmap, so they do automatically get calculated/set for most users.

Reply:
Excluding hapmap...

Reply:
I mean, we compute the centromeres when reading hapmap, right? What happens when we start using some other (non dreadful) interchange format for recombination maps? Duplicate the code that's in hapmap?

Reply:
I don't think we "compute" any telomeres in HapMap files. We just allow them to start at a position > 0 and end at a position < sequence_length. Internally we store these as zero rate regions, which seems reasonable, I guess. It also allows us to "slice" a map while not altering the coordinate space (i.e. "trim=False"), and replace the sliced-out regions with zero rates, while still maintaining the mean rate in the region we keep. So it seems quite handy actually.

Reply:
I also prefer the start_position/end_position, because this is explicit about how the chromosome coordinates differ from the rate map's coordinates. I can imagine these values being useful to rate map consumers, even if they are (mostly) unused internally. E.g. https://github.com/popsim-consortium/stdpopsim/issues/701 (although we have other issues in stdpopsim too, as our chromosome is actually shorter than the recombination map).

Reply:
You could use `slice(trim=True)` to get down to the size of the recombination map. Which makes me thing, actually, should the `start` and `end` of slice default to the rate_map `.start_position` and `.end_position` respectively, rather than `0` and `sequence_length` as they do now? At the moment if we call slice with no parameters, it will reset the start and end positions, which seems a bit counterintuitive.